src/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+set(CMAKE_ASM_NASM_OBJECT_FORMAT "elf")
+enable_language(ASM_NASM)
+
+include_directories(${INCLUDEOS_ROOT}/api/sys)
+include_directories(${INCLUDEOS_INSTALL_DIR}/libcxx/include)
+include_directories(${INCLUDEOS_INSTALL_DIR}/newlib/include)
+include_directories(${INCLUDEOS_ROOT}/src/include)
+include_directories(${INCLUDEOS_ROOT}/api)
+include_directories(${INCLUDEOS_ROOT}/mod/GSL/)
+
+add_custom_command(
+	OUTPUT apic_boot.o
+	COMMAND nasm -f bin -o apic_boot.bin ${CMAKE_CURRENT_SOURCE_DIR}/hw/apic_boot.asm
+	COMMAND objcopy -I binary -O elf32-i386 -B i386 apic_boot.bin apic_boot.o
+	DEPENDS hw/apic_boot.asm
+)
+
+# TODO: i wanted to use a glob, but then found out that not everything is included
+set(OS_OBJECTS  kernel/start.asm kernel/kernel_start.cpp kernel/syscalls.cpp
```

Comment:
(sorry for the late reply, I've been hopping between conferences)

sounds good! so far I'm mostly in a "trying get a minimal CMake conversion up-and-running" mode --- the missing step is adding more validations that the cmake build system creates actually working images.

I will look into this, as soon as the initial version is merged. If you want to, you can add a patch doing that to the pull request (and I would include that change to my pull request).